### PR TITLE
backport: merge bitcoin#23909, #23778, #24495,#25313, #24042, #24057, #24348, #24484, #24489, #24599, #24503, #24506, #24573, #25779, #24736, #24842, #25006, #25076, #25490, #25558 (guix backports: part 2)

### PR DESCRIPTION
**Pull request by kittywhiskers**
## Additional Information

* Depends on https://github.com/dashpay/dash/pull/5355
* Dependency for https://github.com/dashpay/dash/pull/5426

### CODE REVIEW

**Comment by kittywhiskers** [comment]
Build output:

```
4c3257d567e681adc5b4e19603a5092ba4d52a4cfe65b019e714e108aef2c51a  guix-build-306411e74dea/distsrc-306411e74dea-aarch64-linux-gnu/src/dashd
9185afe60bfc647306e04f3ceae734b4ae5232349dcb64bc8c3c86b26012ee14  guix-build-306411e74dea/distsrc-306411e74dea-arm-linux-gnueabihf/src/dashd
58cce33c0ed29b81480e8334f72fc8f85c076afe9e6e1dc30daabf6f6daf1d92  guix-build-306411e74dea/distsrc-306411e74dea-riscv64-linux-gnu/src/dashd
326b5f1e260778430dcaf3f549332d4998038e9716f3d7e9e662db7d470d67b8  guix-build-306411e74dea/distsrc-306411e74dea-x86_64-apple-darwin18/src/dashd
a4f22b3571c746e5abee83550c84212bd5a3c562c97e9e5be92b62e064d3efdd  guix-build-306411e74dea/distsrc-306411e74dea-x86_64-linux-gnu/src/dashd
52a5cf9582d2a84b000f5ed649c13a5265d9b0a7ea3039de9550912639a77388  guix-build-306411e74dea/distsrc-306411e74dea-x86_64-w64-mingw32/src/dashd.exe
b12fbe695d8536ea91d4a8b8c4fc140206271a525e9a78491156e333ddad65bd  guix-build-306411e74dea/distsrc-306411e74dea-aarch64-linux-gnu/src/dash-cli
7e8f9f1e22a9b58778391d82d20d399ef890f156787f60514dda22c6037c808e  guix-build-306411e74dea/distsrc-306411e74dea-arm-linux-gnueabihf/src/dash-cli
5f064c7a9d322191418de7ad8a16411329fa522c154773a69766c44a0f0dff65  guix-build-306411e74dea/distsrc-306411e74dea-riscv64-linux-gnu/src/dash-cli
82a80d5e2ed0722a6d722a65fa177b46e9aa11da1c5d0f0a607096b97e2be58c  guix-build-306411e74dea/distsrc-306411e74dea-x86_64-apple-darwin18/src/dash-cli
780cf369070894f44f9872764d68ac7e46743480071c9ea185cac0d14ae2e0d0  guix-build-306411e74dea/distsrc-306411e74dea-x86_64-linux-gnu/src/dash-cli
589aebb3cae55011d0facc086dee8176f889289d53817c1155df1cad32fd5273  guix-build-306411e74dea/distsrc-306411e74dea-x86_64-w64-mingw32/src/dash-cli.exe
254951fa4ac1fbf474f4f592a769d4aeb8b34ef43322486a69085121b5f312b5  guix-build-306411e74dea/distsrc-306411e74dea-aarch64-linux-gnu/src/dash-tx
986c94fa99ac6c916a62edd8eb3cbd921dd4d6999cb4b0da83967d7bf9c26b80  guix-build-306411e74dea/distsrc-306411e74dea-arm-linux-gnueabihf/src/dash-tx
3f4ca4b55818ededbdd1384eec61ccee4c4a474e5672db85ce75771f7002a391  guix-build-306411e74dea/distsrc-306411e74dea-riscv64-linux-gnu/src/dash-tx
158c6bfaba529dfc95305016a24845adf85d7afa8d5a08814ec650608f3bdcfb  guix-build-306411e74dea/distsrc-306411e74dea-x86_64-apple-darwin18/src/dash-tx
87ca0255381badbd281be1de27dd2b0ebcfd37281f1378245b5e058bd4035a93  guix-build-306411e74dea/distsrc-306411e74dea-x86_64-linux-gnu/src/dash-tx
5618fcb508288d524b344ce0b4e95fe93d4899f41a53d9bc08b5504f5924906f  guix-build-306411e74dea/distsrc-306411e74dea-x86_64-w64-mingw32/src/dash-tx.exe
c5744c0827f06ffe3bbc94a3cb3620844c436d6348f7cf10a56252223aa53578  guix-build-306411e74dea/distsrc-306411e74dea-aarch64-linux-gnu/src/dash-wallet
04ca023aa89acb6672fe87f83908f36255da3d1853914a97296c94624f7caab6  guix-build-306411e74dea/distsrc-306411e74dea-arm-linux-gnueabihf/src/dash-wallet
77b71fa82f580ed2ef2a5102c8cbf6ec1c143bb55d24bfc7bd9abbcbaf60b056  guix-build-306411e74dea/distsrc-306411e74dea-riscv64-linux-gnu/src/dash-wallet
203d9dd88b041d3fe3d60123ce553d265ee026bd534dbe397618f62dd207405a  guix-build-306411e74dea/distsrc-306411e74dea-x86_64-apple-darwin18/src/dash-wallet
6d2aa74b6a9e4ae60df390e60df8a1cc6168b5064324b7da68b525e12520d668  guix-build-306411e74dea/distsrc-306411e74dea-x86_64-linux-gnu/src/dash-wallet
777697790f8daa6c348544c3565a972223fbb8bf18935099ffe2c24b675a6cf9  guix-build-306411e74dea/distsrc-306411e74dea-x86_64-w64-mingw32/src/dash-wallet.exe
ec524dd18b82c21816294ccb383e7c85a7250767aa1d223c88faeb29e0f516b1  guix-build-306411e74dea/distsrc-306411e74dea-aarch64-linux-gnu/src/qt/dash-qt
b93dffcab18d0f0885628fc0e3ad286d4846fdb3f2ba8e785b99fc7e125cf9e4  guix-build-306411e74dea/distsrc-306411e74dea-arm-linux-gnueabihf/src/qt/dash-qt
07372cc5d53ec7fbee6cd825528967035ef3b1cbb1e065380b5285997429f5ac  guix-build-306411e74dea/distsrc-306411e74dea-riscv64-linux-gnu/src/qt/dash-qt
e6108dd9eb1e13894a643922a4de31aa22f4982e3c8dfb5dacea81f78c505cd1  guix-build-306411e74dea/distsrc-306411e74dea-x86_64-apple-darwin18/src/qt/dash-qt
d1797052703a83a4a7f4fb77591700046ac8dbe13961eb96fab22677a1d6f0bf  guix-build-306411e74dea/distsrc-306411e74dea-x86_64-linux-gnu/src/qt/dash-qt
995b8626bca071824c5718bc4ad0f20f033b0893882d12b67ef0c0ef8945232b  guix-build-306411e74dea/distsrc-306411e74dea-x86_64-w64-mingw32/src/qt/dash-qt.exe
1bbaeda142fdb20f749ca47d6e8de2c42983cc6dbfc2180bd8710d25984690ba  guix-build-306411e74dea/distsrc-306411e74dea-aarch64-linux-gnu/src/test/test_dash
1651d9362c5a60fde9c7b71a224ab5687006625c719445a0ea0aee19b1e538be  guix-build-306411e74dea/distsrc-306411e74dea-arm-linux-gnueabihf/src/test/test_dash
9c736d805a363f33b06c334e553ec1976f3b197ee52f837dd853da075220a190  guix-build-306411e74dea/distsrc-306411e74dea-riscv64-linux-gnu/src/test/test_dash
cf43e9e3f390fac78dcb84c68bcade765abe5730c3ce528ade519b2b5f7f47df  guix-build-306411e74dea/distsrc-306411e74dea-x86_64-apple-darwin18/src/test/test_dash
3940e060119c77f9d9578f7947f1f2c3fda4668b372ac69b39cc22f5bd5d966d  guix-build-306411e74dea/distsrc-306411e74dea-x86_64-linux-gnu/src/test/test_dash
3fc3b1bfba589352f78a5c0d97e5799f6e01f665077eaea7a2d6a840c5140dd5  guix-build-306411e74dea/distsrc-306411e74dea-x86_64-w64-mingw32/src/test/test_dash.exe
```

**Review by PastaPastaPasta — DISMISSED** [review]
commit review utACK for merging via merge commit; doing aarch64 guix build now

**Review by knst — CHANGES_REQUESTED** [review]

**Review by UdjinM6 — COMMENTED** [review]
Hashes match and gitian still works.

side note: let's use checksum scripts which would let us all have the same output (and which would match guix github workflow results too) i.e.:
```
# for binaries
( \
SRC_PATH_PREFIX=guix-build-$(git rev-parse --short=12 HEAD)/distsrc- && \
sha256sum ${SRC_PATH_PREFIX}*/src/dash{d,-cli,-tx,-wallet}{,.exe} && \
sha256sum ${SRC_PATH_PREFIX}*/src/qt/dash-qt{,.exe} && \
sha256sum ${SRC_PATH_PREFIX}*/src/test/test_dash{,.exe} \
) | LC_ALL=C sort -k 2
# for archives
find guix-build-"$(git rev-parse --short=12 HEAD)"/output -type f | grep -v SHA256 | xargs sha256sum | LC_ALL=C sort -k 2
```

File: contrib/guix/manifest.scm [hunk 1]
```diff
@@ -16,7 +16,6 @@
              (gnu packages gcc)
              (gnu packages gnome)
              (gnu packages image)
-             (gnu packages imagemagick)
```
[the remark this answers is not in the document]
Comment (UdjinM6) [reply contrib/guix/manifest.scm:19]: Good catch! 👍

[the remark this answers is not in the document]
Comment (kittywhiskers) [reply contrib/guix/manifest.scm:19]: Addressed in 0ca3fa60a1d103643e2722a377c18581f8a707d1

File: doc/dependencies.md [hunk 2]
```diff
@@ -17,6 +17,7 @@ These are the dependencies currently used by Dash Core. You can find instruction
 | libevent | [2.1.11-stable](https://github.com/libevent/libevent/releases) | [2.0.21](https://github.com/bitcoin/bitcoin/pull/18676) | No |  |  |
 | libnatpmp | git commit [4536032...](https://github.com/miniupnp/libnatpmp/tree/4536032ae32268a45c073a4d5e91bbab4534773a) |  | No |  |  |
 | libpng |  |  |  |  | [Yes](https://github.com/dashpay/dash/blob/develop/depends/packages/qt.mk) |
+| Linux Kernel |  N/A | 3.2.0 |
```
[the remark this answers is not in the document]
Comment (kittywhiskers) [reply doc/dependencies.md]: Addressed in da5e6b4e4badc412247c446db463b6ef56ffb810

**Comment by kittywhiskers** [comment]
Added two commits instead of integrating them into backport PR and haven't rebased upon `develop` to avoid re-doing Guix builds locally and in CI

**Review by knst — APPROVED** [review]
LGTM

**Review by UdjinM6 — APPROVED** [review]
confirmed that osx gitian still works

ACK

**Review by PastaPastaPasta — APPROVED** [review]
light-ACK for merging via merge commit

**Pull request merged**